<?xml version="1.0"?>
<launch>
    <param name="realworld" type="string" value="false" />

    <!--Plane Segmentation-->
    <param name="min_x" type="double" value="0"   />
    <param name="max_x" type="double" value="4.0" />
    <param name="min_y" type="double" value="-0.7"/>
    <param name="max_y" type="double" value="0.5" />
    <param name="min_z" type="double" value="-0.4"/>
    <param name="max_z" type="double" value="0.5" />
    <param name="distance"    type="double" value="0.05" />
    
    <node pkg="sensor_fusion" type="camera_seg_plane" name="camera_seg_plane" output="screen">
        <remap from="/cloud" to="/zed0/ds_cloud" />
        <remap from="/output/pickup"  to="/camera/pickup" />
        <remap from="/output/plane"   to="/camera/plane" />
        <remap from="/output/outlier" to="/camera/outlier" />
    </node>

    <!--Circle Detection-->
    <node pkg="sensor_fusion" type="camera_seg_circle" name="camera_seg_circle" output="screen">
        <remap from="/cloud"            to="/camera/outlier" />
        <remap from="/output/reference" to="/camera/reference" />
        <remap from="/output/search"    to="/camera/search" />]
        <remap from="/output/candidate" to="/camera/candidate" />
        <remap from="/output/centroid"  to="/camera/centroid" />
        <remap from="/output/average"   to="/camera/average" />
    </node>

    <!--Transform PointCloud from "/zed/left_camera" to "zed/center"-->
    <node pkg="sensor_fusion" type="laser_transform_pointcloud" name="zed_transform_pointcloud">
        <param name="target_frame" type="string" value="/zed0/zed_center" />
        <remap from="/cloud"    to="/camera/average" />
        <remap from="/cloud/tf" to="/camera/average/tf" />
    </node>
</launch>
